client.go

```diff
@@ -90,6 +97,7 @@ type DefaultClient struct {
 	Scopes        []string `json:"scopes"`
 	Audience      []string `json:"audience"`
 	Public        bool     `json:"public"`
+	MayAct        []string `json:"may_act"`
```

Comment:
> To me, it is not clear if may_act allows another client to delegate to, or the other way around.

The may_act value(s) of a client define which clients are allowed to exchange the tokens of that client. I'm not sure if that is what you mean.

> I assume that may_act was chosen based on this new claim but to me it looks like that claim would be expected in the subject_token?

Correct, may_act was chosen based on the new claim. `delegatable_by` or `impersonatable_by` would bind it semanticaly to one of the exchange modes (impersonation or delegation) while according to the specification it can be used for both.

> The claim might be used, for example, when a subject_token is
presented to the token endpoint in a token exchange request and may_act claim in the subject
token can be used by the authorization server to determine whether the client (or party
identified in the actor_token) is authorized to engage in the requested delegation or
impersonation.

You are completely right, according to the specification this new claim would be expected in the subject_token.
As I tried to explain in https://github.com/ory/hydra/issues/1218 the claim representation described in the specification does not allow to provide more than one authorized actor within the subject token (please correct me if I'm wrong), something that is necessary for enabling delegation chains as implicitly stated e.g. in the description of the actor claim.

The other aspect why we decided to evaluate this internally over the client was, on the one side the fact that we need anyway to add that information to the clients, but also because we found reasonable to allow clients to revoke a a previously given permission to some client to exchange its tokens.